huaweicloud/services/gaussdb/resource_huaweicloud_gaussdb_opengauss_database.go

```diff
@@ -0,0 +1,314 @@
+}
+
+func buildCreateOpenGaussDatabaseBodyParams(d *schema.ResourceData) map[string]interface{} {
+	bodyParams := map[string]interface{}{
+		"name":          d.Get("name"),
+		"character_set": utils.ValueIgnoreEmpty(d.Get("character_set")),
+		"owner":         utils.ValueIgnoreEmpty(d.Get("owner")),
+		"template":      utils.ValueIgnoreEmpty(d.Get("template")),
+		"lc_collate":    utils.ValueIgnoreEmpty(d.Get("lc_collate")),
+		"lc_ctype":      utils.ValueIgnoreEmpty(d.Get("lc_ctype")),
+	}
+	return bodyParams
+}
+
+func resourceOpenGaussDatabaseRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	var (
+		httpUrl = "v3/{project_id}/instances/{instance_id}/databases"
+		product = "opengauss"
+	)
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating GaussDB client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	getBasePath := client.Endpoint + httpUrl
+	getBasePath = strings.ReplaceAll(getBasePath, "{project_id}", client.ProjectID)
+	getBasePath = strings.ReplaceAll(getBasePath, "{instance_id}", instanceId)
+
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	// actually, offset is pageNo in API, not offset
+	var offset int
+	var database interface{}
+	databaseName := d.Get("name").(string)
+
+	for {
+		getPath := getBasePath + buildOpenGaussDatabaseQueryParams(offset)
+		getResp, err := client.Request("GET", getPath, &getOpt)
+		if err != nil {
+			return common.CheckDeletedDiag(d, err, "error retrieving GaussDB openGauss database")
+		}
+		getRespBody, err := utils.FlattenResponse(getResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		database = utils.PathSearch(fmt.Sprintf("databases[?name=='%s']|[0]", databaseName), getRespBody, nil)
+		if database != nil {
+			break
+		}
+		totalCount := utils.PathSearch("total_count", getRespBody, float64(0)).(float64)
+		if int(totalCount) <= (offset+1)*100 {
+			break
+		}
+		offset++
+	}
+	if database == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "error retrieving GaussDB openGauss database")
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", instanceId),
+		d.Set("name", utils.PathSearch("name", database, nil)),
+		d.Set("owner", utils.PathSearch("owner", database, nil)),
+		d.Set("character_set", utils.PathSearch("character_set", database, nil)),
+		d.Set("lc_collate", utils.PathSearch("collate_set", database, nil)),
+		d.Set("lc_ctype", utils.PathSearch("datctype", database, nil)),
+		d.Set("size", utils.PathSearch("size", database, nil)),
+		d.Set("compatibility_type", utils.PathSearch("compatibility_type", database, nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildOpenGaussDatabaseQueryParams(offset int) string {
+	return fmt.Sprintf("?limit=100&offset=%v", offset)
```

Comment:
should it be like this?
```suggestion
	return fmt.Sprintf("?limit=100&offset=%v", offset * 100)
```